Actions/AL-Go-Helper.ps1

```diff
@@ -1784,26 +1810,13 @@ Function AnalyzeProjectDependencies {
         $project = $_
         Write-Host "- Analyzing project: $project"
 
-        # Read project settings
-        $projectSettings = ReadSettings -baseFolder $baseFolder -project $project
+        $projectSettings = ReadSettings -project $project -baseFolder $baseFolder
+        ResolveProjectFolders -baseFolder $baseFolder -project $project -projectSettings ([ref] $projectSettings)
 
-        # Filter out app folders that doesn't contain an app.json file
+        # App folders are relative to the AL-Go project folder. Convert them to relative to the base folder
+        $projectPath = Join-Path $baseFolder $project
         $folders = @($projectSettings.appFolders) + @($projectSettings.testFolders) + @($projectSettings.bcptTestFolders) | ForEach-Object { 
-            $projectFolder = Join-Path $project $_
-            if (!(Test-Path $projectFolder -PathType Container)) {
-                Write-Host "::Warning::Folder $_ doesn't exist, skipping."
-            }
-            elseif (!(Test-Path (Join-Path $projectFolder 'app.json'))) {
-                Write-Host "::Warning::Folder $_ doesn't contain an app.json file, skipping."
-            }
-            else {
-                Resolve-Path $projectFolder -Relative
-            }
-        }
-        # Default to scanning the project folder if no app folders are specified
-        if (-not $folders) {
-            Write-Host "No apps or tests folders found for project $project. Scanning for apps in the project folder."
-            $folders = @(Get-ChildItem -Path (Join-Path $baseFolder $project) -Recurse | Where-Object { $_.PSIsContainer -and (Test-Path (Join-Path $_.FullName 'app.json')) } | ForEach-Object { $_.FullName.Substring($baseFolder.Length+1) } )
+            return (Resolve-Path (Join-Path $projectPath $_) -Relative)
```

Comment:
Using -relative assumes that the current folder is the basefolder right?
We should probably ensure that (like in the resolveprojectfolders)

Reply:
Added "Push-Location Pop-Location" pattern.